build.gradle

```diff
@@ -94,6 +96,17 @@ configure(projectsWithFlags('java')) {
         testCompile 'org.assertj:assertj-core'
         testCompile 'org.mockito:mockito-core'
         testCompile 'org.apache.httpcomponents:httpclient'
+
+        // Fail the build if Java 9+ APIs are used.
+        signature 'org.codehaus.mojo.signature:java18:1.0@signature'
+    }
+
+    animalsniffer {
```

Comment:
If I'm not mistaken, JDK9+ added the `--release` flag which prevents compiling against newer APIs.

http://openjdk.java.net/jeps/247

Reply:
I'm curious if we can still use the JDK9+ APIs selectively using the `--release` option, i.e. use faster implementation in JDK 9/10 if possible, or fall back to less efficient one.

Reply:
Yeah it does allow multi release jars too. Though it might just increase jar size without that much benefit, e.g., eventually they'll optimize string concatenation in newer Java, buy haven't heard of it happening yet

Reply:
Updated to use `--release` option. We now reject the build if the current JDK is older than 9, because JDK 8 javac does not recognize the `--release` option.

Reply:
Realized the we can just remove the `--release` option for JDK 8. Updated again.